### ISSUE TITLE
HACL_Setup: creating database tables fails in RHEL 7

### ISSUE THREAD

**Issue by hermannschwaerzlerUIBK** [HaloACL/HaloACL#14]
When I try to create the necessary database tables for HaloACL by running 
`php HACL_Setup.php`
I get the following error:
```
A database query error has occurred.
Query: SELECT DISTINCT user_id, group_id, group_name
                                        FROM `user`
                                        LEFT JOIN  `halo_acl_group_members` ON  `halo_acl_group_members`.child_id = `user`.user_id
                                        LEFT JOIN `halo_acl_groups` ON `halo_acl_groups`.group_id = `halo_acl_group_members`.parent_group_id
                                        WHERE `user`.user_id = 0 AND `halo_acl_group_members`.child_type = 'user'
                    
Function: DatabaseBase::query
Error: 1146 Table 'uibkwikiHaloACL.halo_acl_group_members' doesn't exist (localhost)
```
This happens during the initialization of the whole "system" when 
`require_once "$mediaWikiLocation/maintenance/commandLine.inc";`
is called in `HACL_Setup.php`.

The software versions I am using:
* Red Hat Enterprise Linux Server release 7.0 (Maipo)
* mysql  Ver 15.1 Distrib **5.5.41-MariaDB**, for Linux (x86_64) using readline 5.1
* PHP 5.4.16
* MediaWiki 1.23.9

**Comment by hermannschwaerzlerUIBK** [comment]
Maybe it's Maria-DB that causes the issue?
On the other hand I am not sure why this **works** for others!

Anyway I was able to "solve" the problem through the following change in the code:
```patch
diff --git a/storage/HACL_StorageSQL.php b/storage/HACL_StorageSQL.php
index ff66efd..0dbbcb4 100644
--- a/storage/HACL_StorageSQL.php
+++ b/storage/HACL_StorageSQL.php
@@ -610,9 +610,13 @@ class HACLStorageSQL {
                                LEFT JOIN $gt ON $gt.group_id = gmt.parent_group_id
                                        WHERE gmt.child_id = $memberID AND gmt.child_type = 'group'";
                }
-               $res = $db->query($sql);
 
                $curGroupArray = array();
+               try {
+                       $res = $db->query($sql);
+               } catch (Exception $e) {
+                       return $curGroupArray;
+               }
 
                while ($row = $db->fetchObject($res)) {
                        if (isset($row->group_id) && isset($row->group_name)) {
```

**Comment by hermannschwaerzlerUIBK** [comment]
This issue is solved due to the merging of pull reqeust #15.
